### DISCUSSION TITLE
Particle Injector Artificial Heating

### DISCUSSION THREAD

**Discussion by PierceGiffin** [epochpic/epoch#458] [Q&A]
Dear EPOCH Team,

The system I wish to model consists of two cold (1e5 K) beams of equal parts electrons and positrons (hence neutral) traveling at non-relativistic (0.2c) relative speeds toward each other. To do so, I’ve implemented injectors on the x boundaries to continually introduce new particles into the beam. Figure 1 (attached) is a plot of momenta in the x-direction versus x-position for the initial configuration of the beams. The beams have low px dispersion, drift velocity of 0.2c, and are well-localized in phase space. However, once the injector begins supplying new particles to the beam, the region near the injector appears to heat up unphysically (see Fig. 2). As can be seen in Fig. 3, which shows the number density in the x and y plane, there appears to be a spurious build-up of charge in the first simulation cell along the x-boundaries, which induces a large field that might be responsible for this heating.

Is my implementation of the injector somehow incorrect? (I’m attaching my current input deck if you wish to examine the particular choice of parameters.) Or is this a known behavior, and if so, what are the techniques to mitigate it? Thank you so much for your time and for developing such a powerful framework. Looking forward to hearing from you!

Best,
Pierce
![image](https://user-images.githubusercontent.com/113052305/221275950-f7884676-706a-42f6-9387-67e65b60f6bd.png)
![image](https://user-images.githubusercontent.com/113052305/221276007-eafd9065-8366-4623-a71b-cf8f7bb57c96.png)
![image](https://user-images.githubusercontent.com/113052305/221276046-484aa1d9-3017-46b5-b152-c62c23ab05b1.png)

**Comment by Status-Mirror** [comment]
Hey Pierce,

I'm not seeing an input deck here. Could you re-attach it?

From your description, one thing to note is that even though you're injecting a neutral beam, the electrons and positrons don't overlap completely as their positions are random inside the cell. This will lead to slight charge fluctuations when particle shapes are considered. I expect this effect to be smoothed out if you increase the particles per cell, if it's the cause of what you're seeing.

Even if you had perfect accuracy, I'm not sure what you expect to happen here. If every cell was perfectly neutral, then there would be no current in the system, and no fields would build up in the window. The beams would pass through each other without interacting, unless you're running with collisions switched on? Currently the code does not support electron-positron annihilation.

Cheers,
Stuart

**Reply by PierceGiffin** [reply]
Hi Stuart,

Thank you for your reply. My apologies, the input deck is attached here. 

I understand that small fluctuations in the particle positions will create the effect of some background EM fields. You are correct, I'm studying the two-stream instability for collisionless particles so it's precisely these fluctuations that seed the instabilities I'm looking for.

However, this effect is greatly magnified in the particle injector. So much so that I'm finding myself unable to inject a cold plasma beam. I've found that if I use periodic conditions on the same boundary as the injector, this problem seems to fix itself. But these boundary conditions then interfere with the plasma that I wish to study. I'm somewhat convinced this is an issue with the method used to inject the particles on an open boundary. Any further insights you could provide would be greatly appreciated!

Best,
Pierce


[input_deck.txt](https://github.com/Warwick-Plasma/epoch/files/10945043/input_deck.txt)
